scripts/prepare-open-api-for-sdk/removeOneOfs.ts

```diff
@@ -0,0 +1,527 @@
+import {
+  camelCase,
+  compact,
+  intersection,
+  omit,
+  uniq,
+  upperFirst,
+} from "lodash";
+import { isDeepStrictEqual } from "util";
+import { removeNotUsedSchemas } from "./remove-not-used-schemas";
+import dotenv from "dotenv";
+dotenv.config();
+import colors from "colors";
+import { asyncMap } from "../helpers/asyncMap";
+
+const sleep = async (ms: number) => {
+  return new Promise((resolve) => setTimeout(resolve, ms));
+};
+const addSpacesInTitle = (title: string) =>
+  title
+    .replaceAll(" ", "")
+    .replace(/([A-Z])/g, " $1")
+    .trim();
+const checkProperties = (schema, title, schemas) => {
+  const properties = Object.fromEntries(
+    Object.entries(schema.properties).map((propertyNameAndEntry) => {
+      const [propertyName, entry] = propertyNameAndEntry as any;
+      if (entry?.oneOf) {
+        return [
+          propertyName,
+          removeOneOf(
+            entry,
+            schemas,
+            addSpacesInTitle(title + upperFirst(camelCase(propertyName)))
+          ).schema,
+        ];
+      }
+      if (entry?.properties) {
+        return [propertyName, checkProperties(entry, title, schemas)];
+      }
+      if (entry.items?.oneOf) {
+        return [
+          propertyName,
+          {
+            ...omit(entry.items, "oneOf"),
+            type: "array",
+            items: removeOneOf(
+              entry.items,
+              schemas,
+              addSpacesInTitle(
+                title + upperFirst(camelCase(propertyName)) + "Item"
+              )
+            ).schema,
+          },
+        ];
+      }
+      return [propertyName, entry];
+    })
+  );
+  return { ...schema, properties };
+};
+
+const removeOneOf = (schema, schemas, title) => {
+  if (Array.isArray(schema)) {
```

Comment:
Bardzo się mi nie podoba. Brak spójności jakieś trochę w tym kodzie. Brak formatowania. Sporo sprzężeń tak jak np. w tej map pierwszej niżej. Nazewnictwo i długości funkcji też nie ułatwia czytania. 

Nie wiem może w pierwszej iteracji może tak zostać. Jednak tera ciężko mi to jakoś przejrzeć nawet w kontekście znalezienia tutaj jakichś bugów. W wynikowym pliku JSON są jakieś rzeczy, które (chyba) nie są poprawne. Np. w `ValidationRuleRules` chyba powinno być chociaż jedno properties. Bo `additionalProperties` pewnie php nie wspiera.

Reply:
ValidationRuleRules jest okej,